src/objects.js

```diff
@@ -3095,14 +3085,31 @@ SpriteMorph.prototype.freshPalette = function (category) {
         return menu;
     };
 
-    // primitives:
+    if (category === 'unified') {
+        // In a Unified Palette custom blocks appear following each category,
+        // but there is only 1 make a block button (at the end).
+        blocks = this.categories.reduce((blocks, category) =>
```

Comment:
Oh yes, `flatMap` would work well here, but it doesn't exist in non-Chromium Edge (44) which while no longer supported, got patched a year ago... Not sure what date we should draw the line at...